Ft/23 link backend frontend

tanerochris opened the pull request:
## Description
Merged all existing PRs
#20  #16  #22 

## How Has This Been Tested?
No test is written for the front end , just backend API endpoints. 

## Screenshots (if applicable, else remove this line / section)


## Checklist:
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added necessary inline code documentation
- [ ] I have added tests that prove my fix is effective and that this feature works
- [x ] New and existing unit tests pass locally with my changes

mernxl committed:
feat: add api error classes, implement catch all handlers in middlewares

mernxl committed:
feat(middlewares): add `json` method to `res` object

mernxl committed:
chore(deps): add `bcrypt`, `jsonwebtoken`, and `moment` packages

mernxl committed:
chore(props): add Authentication dependent properties to voxnostra.properties file

mernxl committed:
feat(libs): add authentication helpers, configure constants

mernxl committed:
fix(libs): fix db initialisation logic

mernxl committed:
fix(libs): fix session setup logic

mernxl committed:
refactor(middleware): add console.error logging for caught error

mernxl committed:
feat(libs): add `setUserSession` function to auth-helpers, fix config constant init

mernxl committed:
feat(schema): add password embedded schema, move user.model into user folder

mernxl committed:
feat(schema): add password embedded schema, move user.model into user folder

mernxl committed:
feat(api): add user-authentication api endpoints

mernxl committed:
7-feat(api): swap testapi endpoint with a health-check endpoint

tanerochris committed:
6-docs(): Api design documentation
OpenAPI 3.0 design
Yaml and JSON files

closes #6

mernxl committed:
7-feat(schema): add email validation regex with onFail message

mernxl committed:
7-fix(middleware): fix sending 404 after response sent

- set content type for catchall error response

mernxl committed:
7-fix(api): fix res.session is undefined for reset-password

- workaround for JS Object destructuring inside an if block considered error by codacity

tanerochris committed:
6-docs(enpoints): updated API docs readme
Added instructions to run API endpoint documentation
Added link to external docs

close

tanerochris committed:
Merge branch 'ft/7-user-authentication' into ft/14-backend-project-crud

mernxl committed:
7-fix(api): fix error when password is null on create

tanerochris committed:
14-feat(backend): Project CRUD API implementation

- Updated readme file
- Added schemas and model related to project feature
- Created an all model file to import into database ,prevents previous error from mongoose "cannot overrite model already compiled"
- Added ApiErrorResponse class to manage api errors
- Wrote tests for Project feature endpoints
- Updated setupTests.js with code to run before and after test suits run
- rename auth directory to user to match docs api endpoints

close #14

tanerochris committed:
14-feat(backend): Project CRUD API implementation

- Updated readme file
- Added schemas and model related to project feature
- Created an all model file to import into database ,prevents previous error from mongoose "cannot overrite model already compiled"
- Added ApiErrorResponse class to manage api errors
- Wrote tests for Project feature endpoints
- Updated setupTests.js with code to run before and after test suits run
- rename auth directory to user to match docs api endpoints

close #14

tanerochris committed:
14-feat(backend): Project CRUD API implementation

- Updated readme file
- Added schemas and model related to project feature
- Created an all model file to import into database ,prevents previous error from mongoose "cannot overrite model already compiled"
- Added ApiErrorResponse class to manage api errors
- Wrote tests for Project feature endpoints
- Updated setupTests.js with code to run before and after test suits run
- rename auth directory to user to match docs api endpoints

close #14

tanerochris committed:
14-feat(backend): Project CRUD API implementation

- added jest options --forceExit --detectOpenHandles --maxWorkers=10 to exit jest when all test pass

close #14

tanerochris committed:
14-fix(backend): Project CRUD API implementation

- fix codacy fail error
close #14

tanerochris committed:
14-fix(backend): Project CRUD API implementation

- fix codacy fail error
close #14

mernxl committed:
4-feat(schema): add app schema based on api-docs and endpoints

- add project schema
- add user schema

tanerochris committed:
14-refactor(structure): Rename folders and moved files

- documented project api endpoints
- refactored project structure, renamed folders and moved files raised in #16
- added prettier configuration

partially resolve #14 reviews

tanerochris committed:
14-refactor(structure): Rename folders and moved files

- documented project api endpoints
- refactored project structure, renamed folders and moved files raised in #16
- added prettier configuration

partially resolve #14 reviews

tanerochris committed:
14-refactor(structure): Rename folders and moved files

- documented project api endpoints
- refactored project structure, renamed folders and moved files raised in #16
- added prettier configuration

partially resolve #14 reviews

mernxl committed:
7-fix(middleware): fix sending 404 after response sent

mernxl committed:
7-feat(libs): add isAuthenticated method in auth-helpers, add index file in libs dir

mernxl committed:
7-docs(schema): add documentation to a password schema method

mernxl committed:
7-feat(api): add /user/me path, refactor routes to suit api-docs

cliffordten committed:
13-feat-(html-pages): added basics pages - [login, signup, home, userprofile]

cliffordten committed:
13-fix-(html-pages): fixed codacy check

cliffordten committed:
13-feat-(html-pages): added some user pages - [userOrg, userProject, userSetting]

tanerochris committed:
feat/15-project-crud-frontend

- create, view, list and delete project
- added app.js file to extend nextjs
- refactored model imports to commonjs
- added prop-types

closes #15

tanerochris committed:
feat/15-project-crud-frontend

- create, view, list and delete project
- added app.js file to extend nextjs
- refactored model imports to commonjs
- added prop-types

closes #15

tanerochris committed:
feat/15-project-crud-frontend

- create, view, list and delete project
- added app.js file to extend nextjs
- refactored model imports to commonjs
- added prop-types

closes #15

tanerochris committed:
feat/15-project-crud-frontend

- create, view, list and delete project
- added app.js file to extend nextjs
- refactored model imports to commonjs
- added prop-types

closes #15

tanerochris committed:
merged

tanerochris committed:
merge

tanerochris committed:
merged

tanerochris committed:
23-feat(application): Combining all PRs

merged the different PRs

closes #23

tanerochris committed:
23-feat(application): Combining all PRs

merged the different PRs

closes #23

tanerochris committed:
23-feat(application): Combining all PRs

merged the different PRs

closes #23

mernxl commented:
Hey @tanerochris, I noticed you are using 

```js
// in /api/user/login.js 
...
const User = mongoose.model('User');
...

// use User to do operations
```
Within the api routes without strictly importing all the models on the path to the said route. this isn't too good as you may request a model which is not already registered, so explicitly requesting a model from the file will be better.

mernxl committed:
chore(build): add use babel-eslint as eslint parser

This change enables features on babel like optional chaining, and removes eslint errors related to the features

mernxl committed:
23-refactor: remove redundant jsonResponse middleware

Next internally sets a .json .send methods on the response object

mernxl committed:
23-refactor: remove old schema files no longer in use

mernxl committed:
23-feat(helpers): add isAuthenticated method, refactor to common js

mernxl committed:
23-feat: link up session to pages, remove page from urls

- remove UserHeader, use AppHeader instead

mernxl committed:
23-fix: fix exports on api-errors.js file

mernxl committed:
23-fix: use a unified session getter, over pages needing session

tanerochris commented:
All models are imported/loaded/registered before the app actually runs,
that's done in app.js, using glob module which gets all model the file
paths by pattern. Importing a module by filepath, in case an adjustment is
done to a file path then will mean changing all reference of this path in
the different files it was used unlike loading it and just calling it as I
did.

On Sat, Jun 6, 2020 at 10:07 PM Rodrick Nfinyoh M <notifications@github.com>
wrote:

> Hey @tanerochris <https://github.com/tanerochris>, I noticed you are using
>
> // in /api/user/login.js
> ...const User = mongoose.model('User');
> ...
> // use User to do operations
>
> Within the api routes without strictly importing all the models on the
> path to the said route. this isn't too good as you may request a model
> which is not already registered, so explicitly requesting a model from the
> file will be better.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/BuildForSDG/voxnostra/pull/24#issuecomment-640158005>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACC3JXMOVHA2Z2MZWKEVLD3RVMN7TANCNFSM4NSCJXOQ>
> .
>


-- 
My twitter  @tanerochris
What I've built https://tchizer.com
Developer Circle Buea Lead facebook.com/groups/DevCBuea/
Whatsapp +237 678431904

Commit:
Initial commit

tanerochris committed:
deploy

tanerochris committed:
deploy

tanerochris committed:
deploy
